lib/astutils.cpp

```diff
@@ -2460,7 +2460,7 @@ bool isMutableExpression(const Token* tok)
         return false;
     if (tok->isLiteral() || tok->isKeyword() || tok->isStandardType() || tok->isEnumerator())
         return false;
-    if (Token::Match(tok, ",|;|:|]|)|}"))
+    if (Token::Match(tok, ",|;|:|]|)|}|::"))
```

Comment:
You shouldn't bail on `::`, this will miss variable changes. Instead you should check `astOperand2()`(or perhaps both) when its a `::`:

```cpp
if (Token::Match(tok, "::"))
    return isMutableExpression(tok->astOperand2());
```

Reply:
Given the example above in case of `std :: strncmp` that will pass any of that to this function and my understanding would be that it makes no sense for `std ::` in that case. Same for a type like `std :: string` which I also see.

The case you refer to would be a scoped variable. I wonder if all parts would also be passed as an expression that should be changed.

I have the feeling that namespaces are not handled properly here. I guess it requires some tests to be written.